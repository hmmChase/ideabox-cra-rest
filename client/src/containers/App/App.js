import React from 'react';
import { Route, Switch } from 'react-router-dom';
import IdeaCardForm from '../../components/IdeaCardForm/IdeaCardForm';
import IdeaContainer from '../IdeaContainer/IdeaContainer';
import './App.css';

const App = () => (
  <main className="App">
    <header className="App-header">
      <h1>IdeaBox</h1>
      <IdeaCardForm />
    </header>
    <Switch>
      <Route
        exact
        path="/"
        render={() => (
          <div>
            <IdeaContainer />
          </div>
        )}
      />
      <Route exact path="/ideas" component={IdeaContainer} />
      <Route
        path="/ideas/:id"
        render={({ match }) => <IdeaContainer id={parseInt(match.params.id)} />}
      />
    </Switch>
  </main>
);

export default App;
