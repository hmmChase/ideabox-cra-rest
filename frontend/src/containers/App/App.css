.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 20px 0;
  background-color: #282c34;
}

h1 {
  margin: 0;
  margin-bottom: 20px;
  font-family: 'Play', sans-serif;
  font-size: 2em;
  color: #FFFF99;
}
