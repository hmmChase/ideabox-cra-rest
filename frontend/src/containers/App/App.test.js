import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    wrapper = shallow(<App {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
