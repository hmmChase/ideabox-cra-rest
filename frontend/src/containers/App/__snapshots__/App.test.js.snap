// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <h1>
            IdeaBox
          </h1>
        </header>,
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/ideas"
          />
          <Route
            path="/ideas/:id"
            render={[Function]}
          />
        </Switch>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            IdeaBox
          </h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "IdeaBox",
          },
          "ref": null,
          "rendered": "IdeaBox",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/ideas"
            />,
            <Route
              path="/ideas/:id"
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/ideas",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/ideas/:id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <h1>
              IdeaBox
            </h1>
          </header>,
          <Switch>
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              component={[Function]}
              exact={true}
              path="/ideas"
            />
            <Route
              path="/ideas/:id"
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              IdeaBox
            </h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "IdeaBox",
            },
            "ref": null,
            "rendered": "IdeaBox",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/ideas"
              />,
              <Route
                path="/ideas/:id"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/ideas",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/ideas/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
